authorization.go

```diff
@@ -0,0 +1,204 @@
+package certauth
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/google/uuid"
+)
+
+// AuthorizationCheck provides an interface for checking request authorization
+// programatically.
+// The Allow* methods will be called upon a request to verify that the provided client
+// is authorized to access the requested resource.
+// A nil value will be returned if the AuthorizationCheck wishes to allow access to the
+// resource. Otherwise the returned error should be a description of why the resource
+// is being denied.
+// See the methods for a description of which Allow* method is chosen depending on the
+// request.
+type AuthorizationCheck interface {
```

Comment:
nit: what about `-er` suffix https://golang.org/doc/effective_go#interface-names?
e.g.
```suggestion
type AuthorizationChecker interface {
```

Reply:
I feel like `Check` is perfectly fine as-is, per it's definition:
>  n. A standard for inspecting or evaluating; a test.

But I don't care that much to argue, so I'll change it.